gpMgmt/bin/gppylib/operations/package.py

```diff
@@ -46,6 +46,19 @@ def dereference_symlink(path):
 GPPKG_METADATA_EXTENSION = ".ini"
 GPPKG_METADATA_FILELINE_PATTERM = re.compile(r""".*permission.*type.*size.*sha1.*md5""")
 
+def get_package_file_list(gppkg_package_name):
+    filelist = []
+    gppkg_metadata_file = "{}/{}{}".format(GPPKG_METADATA_DIR, gppkg_package_name, GPPKG_METADATA_EXTENSION)
+    with open(gppkg_metadata_file, "rb") as f:
+        for linenum, line in enumerate(f):
+            if linenum == 0 and line != b'version=v1\n':
```

Comment:
```suggestion
            if linenum == 0 and not line.startswith(b'version=v1'):
```

this file may be a binary file, the only promise is this file starts with `version=v1`. (don't check the new line

and please note, if the file is a binary file. read by line will read the entire file into memory.  if the file is a bad file, the program will OOM.
the common way is to read small data first, then start decode.

```
f = open("", "rb");
d = f.read(10);
if !d.startwith(b"version=v1"):
    emit error
f.close()
f = open("", "rt")
```

but, the current code is sufficient and does not need to be changed

Reply:
THERE'S ONLY TWO THINGS I HATE IN THIS WORLD.

PEOPLE WHO DON'T CARE USERS' INVALID INPUTS AND ... PEOPLE WHO PUT A BINARY FILE INTO `$GPHOME/usr/local/gppkg`  WHICH ENDS WITH `.ini`